### PR TITLE
Adds card form examples

### PR DESCRIPTION
**Pull request by igorp1**
__what__

This PR adds a few examples that should help merchants integrating with the card component understand how to trigger actions and handle them. You can [see how this looks/works here](https://github.com/duffelhq/token-proxy/pull/288#issuecomment-1908462421). I've used this examples to test our component e2e and in the process caught a few bugs which I've addressed on this PR as well. there's one commit for the bug fixes and one for adding the examples.

The latest version is under a carry tag @`3.5.0--canary-5`. We should be able to release this to v`3.5.0` when using save card works (see the [link above](https://github.com/duffelhq/token-proxy/pull/288#issuecomment-1908462421) for context).

### CODE REVIEW

File: examples/card-form-examples/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# card-form-examples
+
+This a set of integration examples for the Duffel Card Component.
+
+## How to use
+
+1. Install with `yarn`
+2. Create a `.env.local` which should include `DUFFEL_API_KEY`, `DUFFEL_API_ORIGIN`.
+3. If your `DUFFEL_API_ORIGIN` is https:localhost also include `NODE_TLS_REJECT_UNAUTHORIZED=0` to `.env.local`
```
Comment (andrejak) [inline examples/card-form-examples/README.md:9]: Should we suggest customers use `https://api.duffel.com` for `DUFFEL_API_ORIGIN` to make ti more obvious (or even include that in the repo?)
If we wanted to be consistent with the top level env file, the variables there are called `DUFFEL_API_URL` and `DUFFEL_API_TOKEN`.

File: examples/card-form-examples/src/components/4_UsingSavedCardExample.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      <input
+        placeholder="paste a saved card id here"
+        value={cardId}
+        onChange={(e) => {
+          setCardId(e.target.value);
+        }}
+      />
+      {cardId.length !== 26 ? (
+        <p>
+          The card component will appear once a valid card Id is added to the
```
Comment (andrejak) [inline examples/card-form-examples/src/components/4_UsingSavedCardExample.tsx:27]:
```suggestion
          The card component will appear once a valid card ID is added to the
```

File: examples/card-form-examples/src/components/2_SavingCardExample.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import { Container } from "./common/Container";
+
+export const SavingCardExample: React.FC<{ clientKey: string }> = ({
+  clientKey,
+}) => {
+  const { actions, triggerSaveCard } = useDuffelCardFormActions();
+  return (
+    <Container title="Save card">
+      <DuffelCardForm
+        tokenProxyEnvironment="development"
```
Comment (andrejak) [inline examples/card-form-examples/src/components/2_SavingCardExample.tsx:15]: If customers are running and checking the examples, would it be better to default to production? Or is it more for our internal testing?
(I guess production doesn't work yet until those iframe variations are published, but might be helpful in the future)

File: src/components/DuffelCardForm/lib/getNewActionsStack.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { DuffelCardFormAction } from "./types";
+
+export function getNewActionsStack(
+  previousActions: DuffelCardFormAction[],
+  newActions: DuffelCardFormAction[]
+): DuffelCardFormAction[] {
+  if (newActions.length <= previousActions.length) {
+    console.warn(
+      "`actions` prop has changed and it's length is smaller than its previous state. This is not supported. It's advised to use the `useDuffelCardFormActions` hook to manage the actions array instead."
```
[the remark this answers is not in the document]
Comment (andrejak) [reply src/components/DuffelCardForm/lib/getNewActionsStack.ts:9]: Also I am seeing this warning in the console on the examples page, not sure if anything needs to be changed there...

**Review by andrejak — APPROVED** [review]
I mentioned an installation issue I had on slack with yarn 4.0.2, maybe we can make that simpler for any customer that needs to test it out? Otherwise I left a couple small comments but overall it looks good to me!

File: examples/card-form-examples/package.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "card-form-examples",
+  "packageManager": "yarn@4.0.2",
+  "scripts": {
+    "dev": "next dev"
+  },
+  "dependencies": {
+    "@duffel/components": "3.5.0--canary-5",
```
[the remark this answers is not in the document]
Comment (igorp1) [reply examples/card-form-examples/package.json:8]: Def not! Thought I had stashed that, will revert before this is merged

File: examples/card-form-examples/src/components/4_UsingSavedCardExample.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        </p>
+      ) : (
+        <>
+          <DuffelCardForm
+            tokenProxyEnvironment="development"
+            clientKey={clientKey}
+            savedCardData={
+              {
+               id: cardId,
+               brand: "visa", // you must get this from the save card response
```
[the remark this answers is not in the document]
Comment (igorp1) [reply examples/card-form-examples/src/components/4_UsingSavedCardExample.tsx:38]: We need the brand to validate the cvc, on the backend, different brands have different lengths, so we need to know.

[the remark this answers is not in the document]
Comment (igorp1) [reply examples/card-form-examples/src/components/4_UsingSavedCardExample.tsx:38]: I wish we didn't need that. But seems like we can't get away without it.

File: examples/card-form-examples/README.md (hunk 1 above)
Comment (igorp1) [reply examples/card-form-examples/README.md:9]: Good shout, I'll make it consistent and add the suggestion!

File: examples/card-form-examples/src/components/4_UsingSavedCardExample.tsx (hunk 2 above)
Comment (igorp1) [reply examples/card-form-examples/src/components/4_UsingSavedCardExample.tsx:27]: Fixed!

File: examples/card-form-examples/src/components/2_SavingCardExample.tsx (hunk 3 above)
Comment (igorp1) [reply examples/card-form-examples/src/components/2_SavingCardExample.tsx:15]: The tests are mostly for us imo. I'll add this as a env variable though, so it's easier to switch between.

File: examples/card-form-examples/src/pages/index.tsx [hunk 7]
```diff
@@ -1,15 +1,15 @@
-import { SavingAndUsingCardExample } from "@/components/5_SavingAndUsingCardExample";
-import { SavingCardExampleWithoutHook } from "../components/1_SavingCardExampleWithoutHook";
-import { SavingCardExample } from "../components/2_SavingCardExample";
-import { CreatingTemporaryCardExample } from "../components/3_CreatingTemporaryCardExample";
-import { UsingSavedCardExample } from "../components/4_UsingSavedCardExample";
-import React from "react";
+
```
Comment (andrejak) [inline examples/card-form-examples/src/pages/index.tsx:1]: Not really important but there is a blank line here that probably doesn't need to be 🤷

File: src/components/DuffelCardForm/lib/useDuffelCardFormActions.ts [hunk 8]
```diff
@@ -1,46 +1,51 @@
...
-  triggerCreateCardForTemporaryUse: () => void;
+  createCardForTemporaryUse: () => void;
 }
 
 /**
- * This hook gives you convinient helpers to read and reason through the DuffelCardForm integration.
- * Add `actions` to the `DuffelCardForm` actions prop and call the functions to trigger the actions you'd like.
- *
- * In the background, this hook is setting the state on actions prop and the component's useEffect hook will trigger the action.
+ * This hook abstracts the ref for convinience and readability.
```
Comment (andrejak) [inline src/components/DuffelCardForm/lib/useDuffelCardFormActions.ts:26]:
```suggestion
 * This hook abstracts the ref for convenience and readability.
```

File: src/components/DuffelCardForm/lib/getNewActionsStack.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import { DuffelCardFormAction } from "./types";
+
+export function getNewActionsStack(
```
Comment (andrejak) [inline src/components/DuffelCardForm/lib/getNewActionsStack.ts:3]: Is this still used?

**Review by andrejak — APPROVED** [review]

File: src/components/DuffelCardForm/lib/getNewActionsStack.ts (hunk 9 above)
Comment (igorp1) [reply src/components/DuffelCardForm/lib/getNewActionsStack.ts:3]: nope, removed it!

File: examples/card-form-examples/src/pages/index.tsx (hunk 7 above)
Comment (igorp1) [reply examples/card-form-examples/src/pages/index.tsx]: Prettier didn't run, but should be cleaned up now!

File: src/components/DuffelCardForm/lib/useDuffelCardFormActions.ts (hunk 8 above)
Comment (igorp1) [reply src/components/DuffelCardForm/lib/useDuffelCardFormActions.ts:26]: Oops, fixed it, thanks!

**Pull request merged**